### PR TITLE
Follow the authentication spec more closely.

### PR DESCRIPTION
**Pull request by Kneesnap**
Theoretically this shouldn't make much of a difference, but I did notice it when I was reading through the npm code. We should trust the API, not our own understanding of the API, just for safety reasons.

Information Source: https://wiki.vg/Protocol_Encryption

### CODE REVIEW

**Comment by Kneesnap** [comment]
After further research, this is actually more important than it would initially seem.

The Mojang API is case-insensitive, theoretically someone could login with a username of the wrong case, and NMP would accept it. Not the end of the world, but should be fixed.

File: src/server/login.js [hunk 1]
```diff
@@ -82,6 +82,7 @@ module.exports = function (client, server, options) {
         }
         // Convert to a valid UUID until the session server updates and does
         // it automatically
+        client.username = profile.name
         client.uuid = profile.id.replace(/(\w{8})(\w{4})(\w{4})(\w{4})(\w{12})/, '$1-$2-$3-$4-$5')
```
Comment (wvffle) [inline src/server/login.js]: You've placed this just between a line and a comment reffering to that line

Comment (Kneesnap) [reply src/server/login.js]: Good point.

**Comment by wvffle** [comment]
LGTM

**Pull request merged**